plugin/src/main/java/org/opensearch/ml/model/MLModelAutoReLoader.java

```diff
@@ -0,0 +1,352 @@
+        searchSourceBuilder.query(queryBuilder);
+
+        SortBuilder<FieldSortBuilder> sortBuilderOrder = new FieldSortBuilder("create_time").order(SortOrder.DESC);
+        searchSourceBuilder.sort(sortBuilderOrder);
+
+        SearchRequestBuilder searchRequestBuilder = new SearchRequestBuilder(client, SearchAction.INSTANCE)
+            .setIndices(ML_TASK_INDEX)
+            .setSource(searchSourceBuilder);
+
+        searchRequestBuilder.execute(ActionListener.wrap(searchResponseActionListener::onResponse, exception -> {
+            log.error("index {} not found, the reason is {}", ML_TASK_INDEX, exception);
+            throw new IndexNotFoundException("index " + ML_TASK_INDEX + " not found");
+        }));
+    }
+
+    /**
+     * get retry times from the index ".plugins-ml-model-reload" by 1 ml node
+     *
+     * @param localNodeId the filter condition to query
+     */
+    @VisibleForTesting
+    void getReTryTimes(String localNodeId, ActionListener<SearchResponse> searchResponseActionListener) {
+        StepListener<IndicesExistsResponse> indicesExistsResponseStepListener = new StepListener<>();
+        isExistedIndex(ML_MODEL_RELOAD_INDEX, indicesExistsResponseStepListener);
+
+        indicesExistsResponseStepListener.whenComplete(indicesExistsResponse -> {
+            if (!indicesExistsResponse.isExists()) {
+                searchResponseActionListener.onResponse(null);
+            }
+        }, indicesExistsResponseStepListener::onFailure);
+
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+        searchSourceBuilder.fetchSource(new String[] { MODEL_LOAD_RETRY_TIMES_FIELD }, null);
+        QueryBuilder queryBuilder = QueryBuilders.idsQuery().addIds(localNodeId);
+        searchSourceBuilder.query(queryBuilder);
+
+        SearchRequestBuilder searchRequestBuilder = new SearchRequestBuilder(client, SearchAction.INSTANCE)
+            .setIndices(ML_MODEL_RELOAD_INDEX)
+            .setSource(searchSourceBuilder);
+
+        searchRequestBuilder.execute(ActionListener.wrap(searchResponse -> {
+            SearchHit[] hits = searchResponse.getHits().getHits();
+            if (CollectionUtils.isEmpty(hits)) {
+                searchResponseActionListener.onResponse(null);
+                return;
+            }
+
+            searchResponseActionListener.onResponse(searchResponse);
+        }, searchResponseActionListener::onFailure));
+    }
+
+    /**
+     * judge whether the index ".plugins-ml-model-reload" existed
+     *
+     * @param indexName index name
+     */
+    void isExistedIndex(String indexName, ActionListener<IndicesExistsResponse> actionListener) {
+        IndicesExistsRequestBuilder indicesExistsRequestBuilder = new IndicesExistsRequestBuilder(
+            client,
+            IndicesExistsAction.INSTANCE,
+            indexName
+        );
+
+        indicesExistsRequestBuilder.execute(ActionListener.wrap(actionListener::onResponse, actionListener::onFailure));
```

Comment:
How about just checking if index exists in cluster metadata ? Refer to PR #717

Reply:
ok, it is a very useful suggestion, I can reduce the complexity of coding implement, thanks.